test/adapter.test.ts

```diff
@@ -25,164 +22,186 @@ const MIN_ADA_POOL_V1_ID_TESTNET =
 const MIN_ADA_POOL_V1_ID_MAINNET =
   "6aa2153e1ae896a95539c9d62f76cedcdabdcdf144e564b8955f609d660cf6a2";
 
-let adapterTestnet: BlockfrostAdapter;
-let adapterMainnet: BlockfrostAdapter;
+let blockfrostAdapterTestnet: BlockfrostAdapter;
+let blockfrostAdapterMainnet: BlockfrostAdapter;
+let maestroAdapterTestnet: MaestroAdapter;
+let maestroAdapterMainnet: MaestroAdapter;
 
 beforeAll(() => {
-  adapterTestnet = new BlockfrostAdapter(
+  blockfrostAdapterTestnet = new BlockfrostAdapter(
     NetworkId.TESTNET,
     new BlockFrostAPI({
       projectId: mustGetEnv("BLOCKFROST_PROJECT_ID_TESTNET"),
       network: "preprod",
     })
   );
-  adapterMainnet = new BlockfrostAdapter(
+  blockfrostAdapterMainnet = new BlockfrostAdapter(
     NetworkId.MAINNET,
     new BlockFrostAPI({
       projectId: mustGetEnv("BLOCKFROST_PROJECT_ID_MAINNET"),
       network: "mainnet",
     })
   );
+  const cardanoNetworkPreprod: Network = "Preprod";
+  maestroAdapterTestnet = new MaestroAdapter(
+    NetworkId.TESTNET,
+    new MaestroClient(
+      new Configuration({
+        apiKey: "MAESTRO_API_KEY_TESTNET",
+        network: cardanoNetworkPreprod,
+      })
+    )
+  );
+  const cardanoNetworkMainnet: Network = "Mainnet";
+  maestroAdapterMainnet = new MaestroAdapter(
+    NetworkId.MAINNET,
+    new MaestroClient(
+      new Configuration({
+        apiKey: "MAESTRO_API_KEY_MAINNET",
+        network: cardanoNetworkMainnet,
+      })
+    )
+  );
```

Comment:
should use `mustGetEnv` like with Blockfrost above so we can run this in CI
also update `.github/workflows/ci.yml` to get env vars from Github secrets